libs/testserver/server.go

```diff
@@ -18,6 +20,9 @@ type Server struct {
 
 	t testutil.TestingT
 
+	fakeWorkspaces map[string]*FakeWorkspace
+	mu             *sync.Mutex
```

Comment:
The mutex doesn't need to be a pointer because the server struct is never copied.

A zero-initialized mutex works as expected: https://pkg.go.dev/sync#Mutex